src/bot.py

```diff
@@ -1,71 +1,76 @@
-import math
-
 from rlbot.agents.base_agent import BaseAgent, SimpleControllerState
+from rlbot.messages.flat.QuickChatSelection import QuickChatSelection
 from rlbot.utils.structures.game_data_struct import GameTickPacket
 
-from util.orientation import Orientation
+from util.ball_prediction_analysis import find_slice_at_time
+from util.drive import steer_toward_target
+from util.sequence import Sequence, ControlStep
 from util.vec import Vec3
 
 
 class MyBot(BaseAgent):
 
-    def initialize_agent(self):
-        # This runs once before the bot starts up
-        self.controller_state = SimpleControllerState()
+    def __init__(self, name, team, index):
+        super().__init__(name, team, index)
+        self.active_sequence: Sequence = None
 
     def get_output(self, packet: GameTickPacket) -> SimpleControllerState:
-        ball_location = Vec3(packet.game_ball.physics.location)
-
-        my_car = packet.game_cars[self.index]
-        car_location = Vec3(my_car.physics.location)
-
-        car_to_ball = ball_location - car_location
-
-        # Find the direction of our car using the Orientation class
-        car_orientation = Orientation(my_car.physics.rotation)
-        car_direction = car_orientation.forward
-
-        steer_correction_radians = find_correction(car_direction, car_to_ball)
-
-        if steer_correction_radians > 0:
-            # Positive radians in the unit circle is a turn to the left.
-            turn = -1.0  # Negative value for a turn to the left.
-            action_display = "turn left"
-        else:
-            turn = 1.0
-            action_display = "turn right"
+        """
+        This function will be called by the framework many times per second. This is where you can
+        see the motion of the ball, etc. and return controls to drive your car.
+        """
 
-        self.controller_state.throttle = 1.0
-        self.controller_state.steer = turn
+        # Start the renderer. Make sure you call end_rendering at the end of this function.
+        self.renderer.begin_rendering()
 
-        draw_debug(self.renderer, my_car, packet.game_ball, action_display)
+        # This is good to keep at the beginning of get_output. It will allow you to continue
+        # any sequences that you may have started during a previous call to get_output.
+        if self.active_sequence and not self.active_sequence.done:
+            return self.active_sequence.tick(packet)
```

Comment:
Rendering is not stopped if there is an active sequence

Reply:
Good catch. I'll see if it's safe to have the framework call begin and end on the bots' behalf, outside this method.

Reply:
Is it also not recommended practice to name your render groups?

Reply:
No, naming a render group is only necessary if you're doing something fancy. If you call begin_rendering() without specifying a name, it uses the automatic unique name for your bot and it'll work fine.

Named render groups are useful if you want something to persist for longer than your normal render loop, or if you need to render more data than can fit in a single render group.